overlord/snapstate/handlers_prereq_test.go

```diff
@@ -545,5 +545,90 @@ func (s *prereqSuite) TestDoPrereqBaseIsNotBase(c *C) {
 	s.state.Lock()
 	defer s.state.Unlock()
 	c.Check(chg.Status(), Equals, state.ErrorStatus)
-	c.Check(chg.Err(), ErrorMatches, `cannot perform the following tasks:\n.*- test \(declared snap base "some-epoch-snap" has unexpected type "app", instead of 'base'\)`)
+	c.Check(chg.Err(), ErrorMatches, `cannot perform the following tasks:\n.*- test \(cannot install snap base "app-snap": unexpected snap type "app", instead of 'base'\)`)
+}
+
+func (s *prereqSuite) TestDoPrereqBaseNoRevision(c *C) {
+	os.Setenv("SNAPD_BASES_CHANNEL", "channel-no-revision")
+	defer os.Unsetenv("SNAPD_BASES_CHANNEL")
+
+	s.state.Lock()
+
+	t := s.state.NewTask("prerequisites", "test")
+	t.Set("snap-setup", &snapstate.SnapSetup{
+		SideInfo: &snap.SideInfo{
+			RealName: "foo",
+			Revision: snap.R(33),
+		},
+		Channel: "beta",
+		Base:    "some-base",
+		Prereq:  []string{"prereq1"},
```

Comment:
is this `Prereq` needed for this particular test? IIUC, it's not needed since this test is about some-base snap missing and not about missing prereqs (which is what the next test TestDoPrereqNoRevision is about)

Reply:
seems the old tests weren't minimal either afaict